pep/003-dependency-namespaces-and-labels.md

```diff
@@ -0,0 +1,508 @@
+When an existing installation cannot be reused, a new installation is created in the specified namespace.
+
+#### Name Selector
+
+When a user installs a bundle, they can tell Porter to use a specific named installation to satisfy a dependency.
+
+
+#### Interface Selector
+
+Bundles can declare that it implements a well-known bundle interface, which encompasses the app installed by the bundle (mysql), the application version (5.7.12), parameters and credentials required by the bundle, and the outputs generated by the bundle (connection string).
+
+
+### Use Case Solutions
+
+For each use case outlined above, how can we use these new match criteria?
+
+* [Solution: Exclusive Resource](#solution-exclusive-resource)
+* [Solution: Global Resource](#solution-global-resource)
+* [Solution: Scoped Resource](#solution-scoped-resource)
+* [Solution: User Provided Resource](#solution-user-provided-resource)
+* [Solution: Polymorophic Resource](#solution-polymorphic-resource)
+
+#### Solution: Exclusive Resource
+
+Use a label that would not exist already so that a new installation is created just for this bundle in the current namespace.
+
+Bundle
+```yaml
+dependencies:
+  requires:
+  - name: redis
+    namespace: "{{ installation.namespace }}"
+    labels:
+      owner: "{{ installation.name }}"
+```
+
+
+#### Solution: Global Resource
+
+Install OLM and do not set a namespace. The Porter Operator is installed into a namespace, and it reuses the global installation of OLM. When a match cannot be found, OLM is installed without a namespace set.
+
+```yaml
+dependencies:
+  requires:
+  - name: olm
+```
+
+* When more than one installation matches, installations in the current namespace are preferred over global installations.
+* Note that this looks like today's dependency definition because global is the default right now.
+
+
+#### Solution: Scoped Resource
+
+The micro-services use a namespace criteria for their dependency on postgres. The first micro-service won't find a match, and installs postgres in the current namespace. The subsequent micro-services reuse that installation.
+
+Share the dependency between application instances in the same namespace
+```yaml
+dependencies:
+  requires:
+  - name: postgres
+    namespace: "{{ installation.namespace }}"
+    labels:
+      app: my-application
+```
+
+The labels let us have multiple applications (which can be comprised of micro-services) installed in a namespace and only share a postgres installation between micro-services within the same application.
+
+
+#### Solution: User Provided Resource
+
+SQL Server is either set up by a Porter bundle or manually because it's existing infrastructure. For existing infra, you define a bundle that just exposes the connection string via an output. The bundle itself doesn't modify any resources.
+
+Bundle
+```yaml
+dependencies:
+  requires:
+  - name: sqlserver
+    reference: azure/sqlserver # Use this bundle if we can't find an existing one
```

Comment:
Bundle should be deployed as OCI package and allow reference to that package using name + tag

Reply:
I'm sorry I'm not understanding what the question is? Porter already supports requiring a dependency by just its OCI reference.